### PR TITLE
Refactor to move API library into core

### PR DESCRIPTION
**Pull request by paulgb**

### CODE REVIEW

File: crates/controller/src/main.rs [hunk 1]
```diff
@@ -193,8 +189,8 @@ async fn reconcile(
                         SessionLivedBackendStatus::default(),
                     )
                     .await?;
-                },
-                _ => ()
+                }
```
Comment (rolyatmax) [inline crates/controller/src/main.rs:192]: Oh interesting - how does this behave without the `,`?

File: crates/controller/src/main.rs [hunk 2]
```diff
@@ -91,7 +91,7 @@ fn get_application_state(pod: &Pod) -> Option<ContainerState> {
 
             if let Some(terminated) = state.terminated.as_ref() {
                 if terminated.reason.as_deref() == Some("Error") {
-                    return Some(ContainerState::Failed)
+                    return Some(ContainerState::Failed);
```
Comment (rolyatmax) [inline crates/controller/src/main.rs:94]: Just curious: does this produce a change in behavior or are semicolons optional on `return` expressions?

Comment (paulgb) [reply crates/controller/src/main.rs:94]: They're optional, but the `cargo clippy` and `cargo fmt` prefer the semicolon in this case.

File: crates/controller/src/main.rs (hunk 1 above)
Comment (paulgb) [reply crates/controller/src/main.rs:192]: The `,` in a `match` statement can be omitted after a `}` or after the last element. It doesn't change the meaning. In this case, `cargo fmt` decided to remove it.

**Review by rolyatmax — APPROVED** [review]
LGTM!

**Pull request merged**